lib/Dialect/TT/IR/TTOpsTypes.cpp

```diff
@@ -573,6 +573,13 @@ mlir::Type LayoutAttr::getScalarElementType() const {
   return elementType;
 }
 
+bool LayoutAttr::isSharded() const {
```

Comment:
I think we need to call this `hasShardedTensorMemoryLayout`, because `isSharded` has a very different meaning for direct to metal path.